tests/domain_test.py

```diff
@@ -0,0 +1,54 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+from typing import NewType, TypeVar
+
+import sciline as sl
+
+T = TypeVar("T")
+
+
+def test_mypy_detects_wrong_arg_type_of_Scope_subclass() -> None:
+    Param = TypeVar('Param')
+    Param1 = NewType('Param1', int)
+
+    class A(sl.Scope[Param, float]):
+        ...
+
+    A[Param1](1.5)
+    A[Param1]('abc')  # type: ignore[arg-type]
```

Comment:
Note that all tests in this file pass at runtime. `mypy` complains if we remove any of the `type: ignore` comments, but also complains if we put too many, since our config ensures that unused ignore comments are not allowed.